### ISSUE TITLE
[Inline scripts] plugins/**/*.php

### ISSUE THREAD

**Issue by C-Lodder** [joomla-projects/joomla-es6#52]
Only 3 files in the layous directory:

- [x] `plugins\editors\codemirror\layouts\editors\codemirror\element.php`
- [x] `plugins\editors\codemirror\layouts\editors\codemirror\init.php`
- [x] `plugins\installer\webinstaller\webinstaller.php`
- [x] `plugins/system/debug/debug.php`
- [x] `plugins/system/stats/layouts/field/data.php`
- [x] `plugins/installer/packageinstaller/tmpl/default.php`
- [ ] `plugins/editors-xtd/readmore/readmore.php`

**Comment by dneukirchen** [comment]
https://github.com/joomla-projects/joomla-es6/blob/dev/plugins/system/debug/debug.php#L461
https://github.com/joomla-projects/joomla-es6/blob/dev/plugins/system/debug/debug.php#L493
https://github.com/joomla-projects/joomla-es6/blob/dev/plugins/system/stats/layouts/field/data.php#L49
https://github.com/joomla-projects/joomla-es6/blob/dev/plugins/installer/webinstaller/tmpl/default.php#L42
https://github.com/joomla-projects/joomla-es6/blob/dev/plugins/installer/webinstaller/tmpl/default.php#L43
https://github.com/joomla-projects/joomla-es6/blob/dev/plugins/installer/packageinstaller/tmpl/default.php#L56
and probably
https://github.com/joomla-projects/joomla-es6/blob/dev/plugins/editors-xtd/readmore/readmore.php#L67

EDIT: added those to the checklist

**Comment by dgrammatiko** [comment]
I'll do the code mirror

**Comment by dneukirchen** [comment]
@dgt41 can you take a look at the readmore xtd-button? Im not sure if the current implementation of xtd-buttons works with multiple editor instances on the same page??? Dont want to mess things up there. Imo `xtd-readmore` [will override its own script options](https://github.com/joomla-projects/joomla-es6/blob/dev/plugins/editors-xtd/readmore/readmore.php#L57) and will insert the readmore in the last editor on the page. Thats also my main problem with all the usage of Joomla::scriptOptions. Sooner or later we will get naming collusions/conflicts without some kind of convention, but thats another topic. 

thx to @mbabker we dont have to mess up his pet project https://github.com/joomla-projects/joomla-es6/pull/50

**Comment by dgrammatiko** [comment]
`XTD-BUTTONS` need a new API, that exposes the functionality (eg a given method) which then can be called either directly for a button element in the page (eg bellow the editor) or by using a shell button inside the editor (eg tinyMCE, codemirror or any other editor button). This getting a little more tricky for the buttons that use a modal and an iframe but I guess we can use the Joomla.Events or another solution to go around it.

So let's introduce an API for these buttons that will work for our given cases (3: direct functionality, modal and modal with an IFRAME)....

**Comment by dgrammatiko** [comment]
We can close the codemirror ones here as we have a PR in the main repo (codemirror is a field and thus it will be a CE): https://github.com/joomla/joomla-cms/pull/19890

**Comment by dneukirchen** [comment]
@dgt41 what about plugins/editors-xtd/readmore/readmore.php dont we have a PR for this in main?

**Comment by dgrammatiko** [comment]
Nope, this needs some API for the xtd-buttons which doesn't exist atm, (something similar to what @Fedik is doing with the toolbar)
